internal/util/apiclient/error_handler.go

```diff
@@ -26,6 +29,11 @@ func HandleErrorResponse(res *http.Response, requestErr error) error {
 		return err
 	}
 
+	version := res.Header.Get("X-Server-Version")
+	if internal.Version != version {
+		log.Warn("Version mismatch! Server and client version are not the same. Please, update your Daytona versions.")
```

Comment:
I suggest
```
Version mismatch detected. CLI is on version X, Daytona Server is on version X. To ensure maximum compatibility, please make sure the versions are aligned.
```

1. It gives more info to the user as to which version should be updated.
2. Sometimes (myb even most of the time) the error might not happen due to a version mismatch so we don't have to push the user to sync them too much.